src/main/java/MorseCodePresenter.java

```diff
@@ -0,0 +1,48 @@
+public class MorseCodePresenter
+{
+    private MorseCodeFrame view;
+    private MorseCodeConverter model;
+
+    public MorseCodePresenter(MorseCodeFrame view, MorseCodeConverter model)
+    {
+        this.view = view;
+        this.model = model;
+    }
+
+    public void translateToEnglish(String expression)
```

Comment:
What is "expression" here?

Reply:
the expression getting translated to English

Reply:
Then why are you calling view.getMorseExpressionText() inside this method?

Reply:
expression is the current contents of the view's getMorseExpressionText method

Reply:
When are these two different values?

Reply:
If the current Morse text happens to be in English (like if user clicks translate to English before changing initial Morse value), the expression is actually English, not Morse, so I simply called it “expression”